Resync `domparsing` from WPT Upstream

Ahmad-S792 opened the pull request:
#### 5669fbed690bf307416ccbfff81b743e43ef4237
<pre>
Resync `domparsing` from WPT Upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=300853">https://bugs.webkit.org/show_bug.cgi?id=300853</a>

Reviewed by NOBODY (OOPS!).

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/6456a0dc2036af09fe9f1a4545a48a43fe69fc9b">https://github.com/web-platform-tests/wpt/commit/6456a0dc2036af09fe9f1a4545a48a43fe69fc9b</a>

* LayoutTests/imported/w3c/web-platform-tests/domparsing/insert_adjacent_html-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/domparsing/insert_adjacent_html.html:
* LayoutTests/imported/w3c/web-platform-tests/domparsing/insert_adjacent_html-001-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/w3c-import.log:
</pre>

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/51cea0a7b0a2698d10bb2b22224a111d56cf0091. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49792

Ahmad-S792 committed:
Resync `domparsing` from WPT Upstream
https://bugs.webkit.org/show_bug.cgi?id=297807
rdar://158968173

Reviewed by Tim Nguyen.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/10e8a1bfb95fcba72889fb067dcdb95cd59cded7

* LayoutTests/imported/w3c/web-platform-tests/domparsing/XMLSerializer-serializeToString-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/domparsing/XMLSerializer-serializeToString.html:
* LayoutTests/imported/w3c/web-platform-tests/domparsing/createContextualFragment-in-detached-xml-document-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/domparser-spurious-attributes-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/domparser-spurious-attributes.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/idlharness.window-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/idlharness.window.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/idlharness.window.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/resources/dummy-xml.xml: Added.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/domparsing/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/domparsing/xml-parse-serialize-roundtrip-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/domparsing/xml-parse-serialize-roundtrip.html: Added.

Canonical link: https://commits.webkit.org/299081@main
